### PR TITLE
add new capability to limit accepted connections

### PR DESCRIPTION
**Pull request by Jim-Lambert-Bose**
add new capability to limit accepted connections

### CODE REVIEW

**Comment by thinkerou** [comment]
@Jim-Lambert-Bose please rebase and squash the pull request for reviewing, thanks!

**Comment by Jim-Lambert-Bose** [comment]
@thinkerou I thought I had done the rebase properly.  The PR only had one commit.  What am I missing

**Comment by thinkerou** [comment]
because some codes doesn’t belong your commit chnanged.

**Comment by Jim-Lambert-Bose** [comment]
Okay.. working on it.

**Comment by thinkerou** [comment]
still have the same problem.

**Comment by appleboy** [comment]
@Jim-Lambert-Bose Please resolve conflicts.

File: gin.go [hunk 1]
```diff
@@ -288,6 +290,42 @@ func (engine *Engine) Run(addr ...string) (err error) {
...
+	*net.TCPListener
+}
+
+func (ln tcpKeepAliveListener) Accept() (net.Conn, error) {
+	tc, err := ln.AcceptTCP()
+	if err != nil {
+		return nil, err
+	}
+	tc.SetKeepAlive(true)
+	tc.SetKeepAlivePeriod(3 * time.Minute)
```
Comment (thinkerou) [inline gin.go]: why is 3? magic number is't good idea.

**Comment by thinkerou** [comment]
> An option could be to extend *Engine so any Run style func could implement it
Look good, only for supporting `RunLimited` and change Engine maybe it's a little expensive.

**Comment by Jim-Lambert-Bose** [comment]
Closing this after more testing on my part. Keep-alive for http/http2 makes it very difficult to determine how many connections to accept.

**Pull request closed without merging**